pkg/storages/s3/folder.go

```diff
@@ -41,6 +42,8 @@ const (
 	UseListObjectsV1         = "S3_USE_LIST_OBJECTS_V1"
 	RangeBatchEnabled        = "S3_RANGE_BATCH_ENABLED"
 	RangeQueriesMaxRetries   = "S3_RANGE_MAX_RETRIES"
+	UploadWithTimeout        = "S3_UPLOAD_WITH_TIMEOUT"
```

Comment:
Why introduce the two settings?

We can add only the `S3_UPLOAD_TIMEOUT` setting. If it is unset or set to zero it basically means that timeout is disabled.